### PR TITLE
Add debug guide for gcp broker

### LINKED ISSUE #887: Add installation documentation for gcp broker
**Problem**
Add installation instructions in knative-gcp and a link from knative documentation as a pointer to alternate broker implementation.

* [x] Introduction to gcp broker and how is it different from channel based one
* [x] Installation, auth configuration: https://github.com/google/knative-gcp/pull/880
* [x] Provide a simple demo: #936 
* [x] How to debug
* [x] link to main README
* [x] add a section about alternate  broker and introduce gcp broker in knative docs

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by liu-cong**
Fixes #887

## Proposed Changes

- 
-
-

**Release Note**

```release-note
NONE
```

**Docs**

cc @capri-xiyue @grantr @yolocs @grac3gao

### CODE REVIEW

File: docs/install/install-gcp-broker.md [hunk 1]
```diff
@@ -122,3 +127,69 @@ receive events from it, just like any Knative Eventing
...
+* Fanout. Fanout continously pull events from decouple topics for all Brokers, applies Trigger filters, and sends events
+  to consumers. For failed deliveries, it sends the events to the corresponding retry topic.
+  * Code: [main.go](https://github.com/google/knative-gcp/blob/master/cmd/broker/fanout/main.go)
+  * Deployment: It a deployment called `broker-fanout` in the `cloud-run-events` namespace.
+* Retry. Retry continously resends events that have failed in delivery to the consumers.
+  * Code: [main.go](https://github.com/google/knative-gcp/blob/master/cmd/broker/retry/main.go)
+  * Deployment: It a deployment called `broker-retry` in the `cloud-run-events` namespace.
+
+
+### Common Issues
```
Comment (capri-xiyue) [inline docs/install/install-gcp-broker.md:159]: Have we already fixed the broker recreation issues? If user delete and then re-create the broker with the same name, user needs to delete the `broker-fanout` pod

Comment (liu-cong) [reply docs/install/install-gcp-broker.md:159]: Yes this has already been fixed.

**Comment by capri-xiyue** [comment]
the build tests complains that 
`ERROR	https://knative.dev/development/eventing/broker-trigger/
Not Found (HTTP error 404)`

**Pull request merged**